### PR TITLE
fix #45334 - Adds tests for gap serialization.

### LINKED ISSUE #45334: add tests for column-gap, row-gap, gap cssText and getPropertyValue serialization
When working on fixing 
https://github.com/WebKit/WebKit/pull/26398
https://bugs.webkit.org/show_bug.cgi?id=271627

I didn't find any tests covering the serialization rules for gap : `window.getComputedStyle()` is already tested, but not the serialization part when setting the attribute.

```
<div id="target" style="row-gap:normal; column-gap:normal">food</div>

const target = document.getElementById('target');
target.style.cssText;
target.style.getPropertyValue('gap');
window.getComputedStyle(target).gap;


# STP 190 (**Buggy**)
cssText:          "row-gap: normal; column-gap: normal;"
getPropertyValue: "normal"
getComputedStyle: "normal normal"


# After the patch (Minibrowser):
cssText:          "gap: normal;"
getPropertyValue: "normal"
getComputedStyle: "normal"


# Firefox 
cssText:          "gap: normal;"
getPropertyValue: "normal"
getComputedStyle: "normal"


# Chrome 
cssText:          "gap: normal;"
getPropertyValue: "normal"
getComputedStyle: "normal"
```

### PR DESCRIPTION
**Pull request by karlcow**
Adds a couple of tests for testing the values when setting gap. 
Fix #45334

### CODE REVIEW

File: css/css-align/gaps/gap-parsing-002.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+test_valid_value("row-gap", "normal");
+test_valid_value("row-gap", "10px");
+
+const div = document.createElement("div");
+const style = div.style;
+
+test(function() {
+  style.cssText = "row-gap: normal; column-gap: normal;";
+  assert_equals(style.cssText, "gap: normal;");
+}, "'row-gap: normal; column-gap: normal;' is serialized to 'gap:normal;'");
```
Comment (dholbert) [inline css/css-align/gaps/gap-parsing-002.html:28]: nit: in the test description (line 28), it'd probably be good to have a space after the colon in `gap:normal` to accurately describe the value we're expecting here.

Specifically here:
```
}, "'row-gap: normal; column-gap: normal;' is serialized to 'gap:normal;'");
                                                                ^^
```

File: css/css-align/gaps/gap-parsing-002.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}, "'column-gap: normal; row-gap: 10px;' is serialized to 'gap: 10px normal;'");
+
+test(function() {
+  style.cssText = "column-gap: normal; row-gap: 10px;";
+  assert_equals(style.getPropertyValue('gap'), "10px normal");
+}, "getPropertyValue for 'column-gap: normal; row-gap: 10px;' returns '10px normal'");
+
+</script>
+</body>
+</html>
```
Comment (dholbert) [inline css/css-align/gaps/gap-parsing-002.html:67]: nit: missing newline at end of file (shown as an angry red circle in github review viewer)

**Review by dholbert — COMMENTED** [review]
Looks good with 2 nits -- minor typo in a test message, and missing newline at end of file.  Mind fixing those?

**Review by nt1m — APPROVED** [review]

File: css/css-align/gaps/gap-parsing-002.html (hunk 1 above)
Comment (karlcow) [reply css/css-align/gaps/gap-parsing-002.html:28]: oopsie. Thanks @dholbert

**Comment by nt1m** [comment]
LGTM with Daniel's comments addressed

File: css/css-align/gaps/gap-parsing-002.html (hunk 1 above)
Comment (karlcow) [reply css/css-align/gaps/gap-parsing-002.html]: fixed.

File: css/css-align/gaps/gap-parsing-002.html (hunk 2 above)
Comment (karlcow) [reply css/css-align/gaps/gap-parsing-002.html]: fixed

**Comment by dholbert** [comment]
Looks good to me, thanks for fixing the nits! --> Pushing the "Enable auto-merge" button.

**Pull request merged**